apps/ledger-live-mobile/src/families/celo/VoteFlow/02-Summary.tsx

```diff
@@ -0,0 +1,464 @@
+import useBridgeTransaction from "@ledgerhq/live-common/bridge/useBridgeTransaction";
+import {
+  formatCurrencyUnit,
+  getCurrencyColor,
+} from "@ledgerhq/live-common/currencies/index";
+import { useValidatorGroups } from "@ledgerhq/live-common/families/celo/react";
+import {
+  CeloValidatorGroup,
+  Transaction,
+} from "@ledgerhq/live-common/families/celo/types";
+import { defaultValidatorGroupAddress } from "@ledgerhq/live-common/families/celo/logic";
+import { AccountLike } from "@ledgerhq/types-live";
+import { Text } from "@ledgerhq/native-ui";
+import { useTheme } from "@react-navigation/native";
+import { BigNumber } from "bignumber.js";
+import invariant from "invariant";
+import React, {
+  ReactNode,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { Trans } from "react-i18next";
+import { Animated, SafeAreaView, StyleSheet, View } from "react-native";
+import Icon from "react-native-vector-icons/dist/Feather";
+import { useSelector } from "react-redux";
+import { TrackScreen } from "../../../analytics";
+import { rgba } from "../../../colors";
+import Button from "../../../components/Button";
+import Circle from "../../../components/Circle";
+import CurrencyIcon from "../../../components/CurrencyIcon";
+import CurrencyUnitValue from "../../../components/CurrencyUnitValue";
+import Touchable from "../../../components/Touchable";
+import { ScreenName } from "../../../const";
+import DelegatingContainer from "../../tezos/DelegatingContainer";
+import { accountScreenSelector } from "../../../reducers/accounts";
+import ValidatorImage from "../ValidatorImage";
+
+type Props = {
+  navigation: any;
+  route: { params: RouteParams };
+};
+
+type RouteParams = {
+  validator: CeloValidatorGroup;
+  transaction?: Transaction;
+  fromSelectAmount: boolean;
+  amount?: number;
+};
+
+export default function VoteSummary({ navigation, route }: Props) {
+  const { validator } = route.params;
+  const { colors } = useTheme();
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+
+  invariant(account, "account must be defined");
+
+  const validators = useValidatorGroups();
+  const mainAccount = getMainAccount(account, parentAccount);
+  const bridge = getAccountBridge(account, undefined);
+
+  const chosenValidator = useMemo(() => {
+    if (validator !== undefined) {
+      return validator;
+    }
+
+    return validators[0];
+  }, [validators, validator]);
+
+  const {
+    transaction,
+    updateTransaction,
+    setTransaction,
+    status,
+    bridgePending,
+    bridgeError,
+  } = useBridgeTransaction(() => {
+    const tx = route.params.transaction;
+
+    if (!tx) {
+      const t = bridge.createTransaction(mainAccount);
+
+      return {
+        account,
+        transaction: bridge.updateTransaction(t, {
+          mode: "vote",
+          recipient: defaultValidatorGroupAddress(),
+          amount: route.params.amount ?? 0,
+        }),
+      };
+    }
```

Comment:
Confirming that all overridden components have generated counterparts.

To answer your second comment, I believe that this is done to prevent the risk of another transaction being accidentally created upon re-render of the component. The pattern is used in a similar way in Cosmos. Maybe @trentkrogers wants to chime in, but I don't see a problem or improved performance removing it.